cmd/access/access_node_builder.go

```diff
@@ -30,171 +33,94 @@ type AccessNodeBuilder struct {
 	unstakedMiddleware      *p2p.Middleware
 }
 
-func FlowAccessNode() *AccessNodeBuilder {
-	return &AccessNodeBuilder{
+func FlowAccessNode() *FlowAccessNodeBuilder {
+	return &FlowAccessNodeBuilder{
 		FlowNodeBuilder: cmd.FlowNode(flow.RoleAccess.String()),
 	}
 }
+func (anb *FlowAccessNodeBuilder) IsStaked() bool {
+	return anb.staked
+}
 
-func (anb *AccessNodeBuilder) Initialize() *AccessNodeBuilder {
-
-	anb.PrintBuildVersionDetails()
+func (anb *FlowAccessNodeBuilder) parseFlags() {
 
 	anb.BaseFlags()
 
 	anb.ParseAndPrintFlags()
-
-	anb.validateParams()
-
-	// for the staked access node, initialize the network used to communicate with the other staked flow nodes
-	if anb.staked {
-		anb.EnqueueNetworkInit()
-	}
-
-	// if an unstaked bind address is provided, initialize the network to communicate on the unstaked network
-	if anb.unstakedNetworkBindAddr != cmd.NotSet {
-		anb.EnqueueUnstakedNetworkInit()
-	}
-
-	anb.EnqueueMetricsServerInit()
-
-	anb.RegisterBadgerMetrics()
-
-	anb.EnqueueTracer()
-
-	return anb
 }
 
-func (anb *AccessNodeBuilder) validateParams() {
-	if anb.staked {
-		return
+func (anb *FlowAccessNodeBuilder) initLibP2PFactory(nodeID flow.Identifier,
```

Comment:
these are 3 util methods that can be used by staked or unstaked AN to create an unstaked network

Reply:
Nit: Those  methods are great, but would deserve a comment per, to describe the span of their work (some of these comments are in `enqueueUnstakedNetworkInit`).

Reply:
yup - added comments to each of those 3 method